scripts/modules/b2b-account/quotes.js

```diff
@@ -132,14 +132,29 @@ define([
             }
             if (isSalesRep) {
                 if (!self.model.get("b2bAccounts")) {
-                    var b2bAccount = new B2BAccountModels.b2bAccounts({ pageSize: 200 });
-                    b2bAccount.apiGet().then(function (accounts) {
-                        self.model.set("b2bAccounts", accounts);
-                        self.render();
+                    $(document).ready(function () {
+                        $("#selectb2bAccount").mzdropdown({
+                            model: B2BAccountModels.b2bAccounts,
+                            pageSize: 20,
+                            displayField: 'companyOrOrganization',
+                            placeHolder: 'Select Account',
+                            filterOption: 'cont'
+                        });
                     });
                 }
-            }
 
+            }
+            $(document).ready(function () {
+                $(document).on('click', '.mozu-dropdown', function () {
+                    $("#createQuoteHompageBtn").prop("disabled", false);
+                });
+                $(document).on('change', '#selectb2bAccount > input', function () {
```

Comment:
I get why you're doing this but this feels a bit icky. Can't we trigger a `change` event manually when the selected value changes? Then you can simply do `$(".mozu-dropdown").on("change", function (value) {` instead of the above.

Also is there no way to set the `value` directly on your root HTML element once an option is selected? Setting value or data-value on the root element would be cleaner and easier to grab that putting it on the input.

Reply:
As this is custom dropdown we added input here in this case we need event from document so this "$(".mozu-dropdown").on("change","  will not work here.